tests/test_symbolic_data.py

```diff
@@ -60,3 +61,18 @@ def test_second_derivatives_space(derivative, dimension, order):
     s_expr = as_finite_diff(u.diff(dimension, dimension), indices)
     assert(simplify(expr - s_expr) == 0)  # Symbolic equality
     assert(expr == s_expr)  # Exact equailty
+
+
+def init(data):
+    data = np.zeros(data.shape)
+
+
+def test_symbol_cache(nx=1000, ny=1000):
+    target_len = len(devito.interfaces._SymbolCache)
+
+    for i in range(10):
+        u = DenseData(name='u', shape=(nx, ny), dtype=np.float64, space_order=2, initializer=init)
+
+        assert(len(devito.interfaces._SymbolCache) == target_len + 1)
```

Comment:
This would make this solution somewhat less temporary.
Also, if we are going to use `weakref`, the `weakref` object will still be stored in the SymbolCache, increasing the length.